lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,448 @@
+      to the system done by the previous one. (LUKS cannot be opened,
+      when it does not exist.)"
+
+author:
+    "Jan Pokorny (@japokorn)"
+'''
+
+EXAMPLES = '''
+- name: create and open the LUKS container
+  luks_device:
+    device: "/dev/loop0"
+    state: "present"
+    open: true
+    key: "/vault/keyfile"
+
+- name: close the LUKS container "mycrypt"
+  luks_device:
+    open: false
+    name: "mycrypt"
+
+- name: add new key to the LUKS container
+  luks_device:
+    device: "/dev/loop0"
+    state: "present"
+    key: "/vault/keyfile"
+    new_key: "/vault/keyfile2"
+
+- name: remove existing key from the LUKS container
+  luks_device:
+    device: "/dev/loop0"
+    remove_key: "/vault/keyfile2"
+
+- name: completely remove the LUKS container and its contents
+  luks_device:
+    device: "/dev/loop0"
+    state: "absent"
+'''
+
+RETURN = '''
+name:
+    description:
+        When 'open' option is used returns (generated or given) name
+        of LUKS container. Returns None if no name is supplied.
+    returned: success
+    type: string
+    sample: "luks-c1da9a58-2fde-4256-9d9f-6ab008b4dd1b"
+'''
+
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+
+RETURN_CODE = 0
+STDOUT = 1
+STDERR = 2
+
+# used to get <luks-name> out of lsblk output in format 'crypt <luks-name>'
+# regex takes care of any possible blank characters
+LUKS_NAME_REGEX = re.compile(r'\s*crypt\s+([^\s]*)\s*')
+# used to get </luks/device> out of lsblk output
+# in format 'device: </luks/device>'
+LUKS_DEVICE_REGEX = re.compile(r'\s*device:\s+([^\s]*)\s*')
+
+
+class CryptHandler(object):
+
+    def __init__(self, module):
+        self._module = module
+        self._module.get_bin_path('cryptsetup', True)
+        self._module.get_bin_path('lsblk', True)
+
+    def _run_command(self, command):
+        return self._module.run_command(command)
+
+    def run_luks_create(self, device, keyfile):
+        # create a new luks container; use batch mode to auto confirm
+        result = self._run_command(['cryptsetup', 'luksFormat',
+                                    '-q', device, keyfile])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while creating LUKS on %s: %s'
+                             % (device, result[STDERR]))
+
+    def run_luks_open(self, device, keyfile, name):
+        result = self._run_command(['cryptsetup', '--key-file', keyfile,
+                                    'open', '--type', 'luks', device, name])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while opening LUKS container on %s: %s'
+                             % (device, result[STDERR]))
+
+    def run_luks_close(self, name):
+        result = self._run_command(['cryptsetup', 'close', name])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while closing LUKS container %s' % (name))
+
+    def run_luks_remove(self, device):
+        self._module.get_bin_path('wipefs', True)
```

Comment:
You also need to store this into a variable, otherwise you can't use the resulting path.

Reply:
Fixed.